### ISSUE TITLE
Partner Sharing Broken After Upgrade to v1.123.0

### ISSUE THREAD

**Issue by jiongxuan** [immich-app/immich#14847]
### The bug

I have encountered two significant issues after upgrading Immich from version v1.122.1 to v1.123.0:

1. Partner Photos Not Visible on Mobile Devices: After upgrading, both my wife and I are unable to see each other’s photos on our Android and iOS devices using the Partner Sharing feature. This functionality worked perfectly in the previous version (v1.122.1).

2. Unable to Re-add Partner After Removal: In an attempt to resolve the issue, I went to the settings and removed each other as partners. However, when trying to re-add a partner by clicking on “Add partner”, the following error message appears:

> Looks like you shared your photos with all users or you don't have any user to share with.

This prevents us from re-establishing the Partner Sharing connection.

### The OS that Immich Server is running on

Synology NAS

### Version of Immich Server

v1.123.0

### Version of Immich Mobile App

v1.123.0

### Platform with the issue

- [X] Server
- [X] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
It's almost identical to the online template and has no changes except for the version.
```

### Your .env content

```Shell
It's almost identical to the online template and has no changes except for the version.
```

### Reproduction steps

1. Upgrade Immich: Upgrade the Immich application from v1.122.1 to v1.123.0.

2. Verify Partner Sharing: On both Android and iOS devices, navigate to the Partner Sharing section. Observe that the partner’s photos are no longer visible.

3. Attempt to Reset Partner Sharing: Go to the settings and remove the existing partner from Partner Sharing. Attempt to add the partner again by clicking on “Add partner”.  Then receive the error message:

> Looks like you shared your photos with all users or you don't have any user to share with.

Unable to re-add the partner.

### Additional information

Previous Version: v1.122.1

**Comment by bo0tzz** [comment]
Do you have the "public users" setting turned off?

**Comment by jiongxuan** [comment]
EDIT: After further investigation, I have made some additional findings regarding the issues with Partner Sharing after upgrading to **v1.123.0**.

1. **Database Restoration and Web Functionality:**
   - I restored the database to a state **before** removing the partner.
   - **On the Web:** Partner photos are now successfully displayed, confirming that the restoration has resolved the issue on the web platform.

2. **Persistent Issues on Mobile Devices (Android and iOS):**
   - **On Android and iOS Devices:**
     - Navigating to **Library -> Partners** still shows an empty list with the message:
       ```
       Your photos are not yet shared with any partner.
       ```
     - When attempting to **Add Partner**, the following error message appears:
       ```
       No more users to add.
       ```
     - This behavior is inconsistent with the Web platform, where adding a partner works as expected.

**Summary of Current Status:**
- **Web Platform:** Partner Sharing is functioning correctly after database restoration.
- **Mobile Platforms (Android and iOS):** Unable to view partner photos and cannot re-add partners due to error messages.

This issue continues to significantly impact our ability to share photos between partners on mobile devices. Any assistance or guidance to resolve the mobile-specific problems would be greatly appreciated.

**Comment by jiongxuan** [comment]
> Do you have the "public users" setting turned off?

@bo0tzz No, it has always been enabled. Partner sharing functions correctly on the Web, but on mobile devices (Android and iOS), we are still unable to view each other’s photos.

**Comment by schuhbacca** [comment]
Can you log out and in on the devices?

**Comment by schuhbacca** [comment]
Can you log out and in on the devices?

**Comment by alextran1502** [comment]
Can you capture the screenshot of this settings on the web in Administration > Settings > Server Settings

<img width="976" alt="image" src="https://github.com/user-attachments/assets/da61d513-2c2e-4b79-929b-6e9135e41231" />

Also, try logout and log back in

**Comment by jiongxuan** [comment]
> Can you capture the screenshot of this settings on the web in Administration > Settings > Server Settings
> 
> 
> 
> <img width="976" alt="image" src="https://github.com/user-attachments/assets/da61d513-2c2e-4b79-929b-6e9135e41231" />
> 
> 
> 
> Also, try logout and log back in

Sure. See below. （I haven't change these option. It's always empty）

![image](https://github.com/user-attachments/assets/2fbd91d4-a291-417b-91b2-10ee634f0349)

**Comment by jiongxuan** [comment]
@alextran1502  Thank you very much! After turning on this option, I successfully saw the partner's photo.

Suggestion: Is it possible to consider adding a prompt on the Partner page? If this option is not turned on and a partner has been added before, it will prompt "The administrator has closed public users", etc. This will facilitate prompting users and avoid some misleading.

Thanks again for your help! Merry Christmas! 🎉

**Comment by alextran1502** [comment]
Good idea! Thanks! Merry Christmas to you as well!
